### DISCUSSION TITLE
How to deal with i18n

### DISCUSSION THREAD

**Discussion by woylie** [ory/kratos#2460] [Q&A] [answered]
What is the best way to handle translations of the messages that Kratos returns? I know the messages are in the modules in the [text folder](https://github.com/ory/kratos/tree/master/text), but those are Go modules and as such not easy to parse.

What I'm missing is either a CLI command that returns all Kratos messages including the error code in some parseable format (e.g. JSON, or even just a tsv or csv), or a file in the Kratos repo in such a format from which the message modules would be generated.

That would allow me to just add that file to my application and use a script or macro to turn it into something I can work with.

One complication with the message modules is the usage of `fmt.Sprintf`, which makes it even harder to parse.

#691 suggested a solution that involved changing the return format for messages on the API level, but it was unfortunately closed as out of scope.

But I don't even care much if I cannot translate the message strings coming from the API dynamically. What I'd really like is a way to retrieve a machine-readable list of codes and messages, so that I can integrate that into my application and generate new translation templates with a single command.

Any thoughts or insights on this topic?

**Comment by aeneasr** [comment, marked as answer]
We could probably repurpose


https://github.com/ory/kratos/blob/b682689dd781d52d164b0c6edaf77ab15c9ef154/cmd/clidoc/main.go#L157

to also return a machine-readable format (e.g. in JSON). Would that help?

**Reply by woylie** [reply]
That's the function that generates https://www.ory.sh/docs/kratos/concepts/ui-user-interface#operation/getVersion, right? Yes, that would be immensely helpful!

**Reply by woylie** [reply]
@aeneasr What do you think should we do with the output of that updated `writeMessages` function? Just add the JSON to the docs? Add a CLI command? Add an API route? Add a (make) command for the dev environment? Something else?
